contracts/TokenDistributor.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {StakingRewardsV2} from "./StakingRewardsV2.sol";
+import {RewardEscrowV2} from "./RewardEscrowV2.sol";
+import {IKwenta} from "./interfaces/IKwenta.sol";
+import {ITokenDistributor} from "./interfaces/ITokenDistributor.sol";
+
+contract TokenDistributor is ITokenDistributor {
+    /// @notice represents the status of if a person already
+    /// claimed their epoch
+    mapping(address => mapping(uint => bool)) public claimedEpochs;
+
+    /// @notice kwenta interface
+    IKwenta public kwenta;
+
+    /// @notice rewards staking contract
+    StakingRewardsV2 public stakingRewardsV2;
+
+    /// @notice escrow contract which holds (and may stake) reward tokens
+    RewardEscrowV2 public rewardEscrowV2;
+
+    /// @notice last recorded balance of KWENTA in contract
+    uint public lastTokenBalance;
+
+    /// @notice last checkpoint time
+    uint public lastCheckpoint;
+
+    /// @notice starting week of deployment
+    uint public startTime;
+
+    /// @notice array for tokens allocated to each epoch
+    uint[1000000000000000] public tokensPerEpoch;
```

Comment:
I took that from Velo because thats what they use for their checkpointing system. I added a gas snapshot and the current deployment cost is 760K gas (not sure if thats a lot). I tried testing other ways to initialize but it just bricks checkpointToken because of the way Velo does it. Thoughts on this @JChiaramonte7?

Reply:
I think you can use a mapping: `mapping(uint256 => uint256)`.
But curious to see if the above offers some gas advantage 🤔